library/core/src/any.rs

```diff
@@ -610,6 +610,84 @@ impl dyn Any + Send + Sync {
 /// While `TypeId` implements `Hash`, `PartialOrd`, and `Ord`, it is worth
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
+///
+/// **Note on Variance**:
+/// Suppose `SubType` is a subtype of `SuperType`,
+/// that is, `SubType` can be used wherever `SuperType` can be used,
+/// and can be freely as-casted to it,
+/// and `CoVar`is a generic data type that has a covariant type parameter.
```

Comment:
```suggestion
/// Say we have a `CoVar<T>` generic type, which happens to be
/// covariant over `T` (for instance, `PhantomData<T>`).
///
```

Reply:
Yeah that is probably simpler, and even an example!